tachyon/math/base/semigroups.h

```diff
@@ -167,18 +167,17 @@ class MultiplicativeSemigroup {
   // return: [c, c * g, c * g², ..., c * g^{|size| - 1}]
   constexpr static std::vector<MulResult> GetSuccessivePowers(
       size_t size, const G& generator, const G& c = G::One()) {
-    std::vector<MulResult> ret;
-    ret.resize(size);
-    size_t num_elems_per_thread =
-        base::GetNumElementsPerThread(ret, kDefaultParallelThreshold);
-    OPENMP_PARALLEL_FOR(size_t i = 0; i < size; i += num_elems_per_thread) {
-      MulResult pow = generator.Pow(i);
-      if (!c.IsOne()) pow *= c;
-      for (size_t j = i; j < i + num_elems_per_thread && j < size; ++j) {
-        ret[j] = pow;
-        pow *= generator;
-      }
-    }
+    std::vector<MulResult> ret = base::CreateVector(size, MulResult::Zero());
```

Comment:
Can you make this change a separate commit?

Reply:
Since we are talking about removing `base::CreateVector()` which is similar to the constructor of `std::vector<>`, I just change it to `std::vector<MulResult> ret(size);`